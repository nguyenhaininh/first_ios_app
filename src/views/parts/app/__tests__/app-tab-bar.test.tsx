/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import AppTabBar, { IAppTabItem } from '../app-tab-bar';

import * as renderer from 'react-test-renderer';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';

it('renders correctly', () => {
  const onPress = jest.fn();
  const history = createMemoryHistory({ initialEntries: ['/tab/feed'] });
  const tree = renderer.create(
    <Router history={history}>
      <AppTabBar
        index={0}
        items={[{
          onPress,
          title: 'Title',
          sourceOff: 123,
          sourceOn: 456
        } as IAppTabItem
        ]}
      />
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
