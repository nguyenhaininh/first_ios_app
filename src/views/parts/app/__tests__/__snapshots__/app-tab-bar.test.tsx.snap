// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#f7f7f7",
      "borderTopColor": "#dddddd",
      "borderTopWidth": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f7f7f7",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": 62,
        }
      }
    >
      <Image
        source={456}
        style={
          Object {
            "marginBottom": 2,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#004098",
            "fontSize": 9,
          }
        }
      >
        Title
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
